### ISSUE TITLE
Throws on iOS 10 

### ISSUE THREAD

**Issue by qwtel** [webcomponents/webcomponentsjs#617]
Including `webcomponents.js` v0.7.22 in Safari on iOS 10.0.1 throws

```
TypeError: undefined is not an object (evaluating 'descriptor.value')
installProperty — webcomponents.js:235
registerInstanceProperties — webcomponents.js:182
registerInternal — webcomponents.js:269
register — webcomponents.js:260
(anonymous function) — webcomponents.js:4365
global code — webcomponents.js:4367
```

The part in question is
```js
var descriptor = getDescriptor(source, name);
var getter, setter;
if (typeof descriptor.value === "function") {
```

and the error happens when `source` is the window object and `name` is `'showModalDialog'` (returns `undefined`).

It can be quick-fixed by changing `getDescriptor` in `ShadowDOM/wrappers.js` to:

```js
  function getDescriptor(source, name) {
    try {
      return Object.getOwnPropertyDescriptor(source, name) || dummyDescriptor;
    } catch (ex) {
      // JSC and V8 both use data properties instead of accessors which can
      // cause getting the property desciptor to throw an exception.
      // https://bugs.webkit.org/show_bug.cgi?id=49739
      return dummyDescriptor;
    }
  }
```

**Comment by tschwartz** [comment]
@qwtel what do you think the ETA is on getting this fixed?

**Comment by phpthug** [comment]
same question as @tschwartz

**Comment by Rbauman** [comment]
ETA and quick fix would be greatly appreciated.

**Comment by qwtel** [comment]
i have a PR open with a quick fix (above). no idea if or when this will make it into the repo. it's a quick fix, not a solution to whatever is causing it.

**Comment by mickeywu** [comment]
Thanks @qwtel, it would be nice to get this issue looked at ASAP.

**Comment by addyosmani** [comment]
I'll comment on the PR.

**Comment by intcreator** [comment]
Does this break if I use webcomponents.min.js as well?  And does this mean that basically every Polymer app is broken (assuming they rely on webcomponents polyfills)?

**Comment by simonmorley** [comment]
As per my comment in the referenced issue, use this to override for a temp. fix

```
  "overrides": {
    "webcomponentsjs": {
      "main": [
        "webcomponents-lite.min.js"
      ]
    }
  },
```

**Comment by Mrboutte** [comment]
@simonmorley not really sure where in the bower.json do you mean to put the patch. Attached a screenshot of how I have it set. I'm tried with your code and specifying the path with to the webcomponents-lite.min.js and the app will still not run on ios10

![image](https://cloud.githubusercontent.com/assets/6290747/19404198/12f48352-923a-11e6-9902-906b3f1563d2.png)

**Comment by uqleadegroot** [comment]
There is no release for this change -ie including web minimised versions and polyfills included in webcomponents.js? What is the timeline for this release?

**Comment by jannakha** [comment]
+1 
is there a release/build planned for this hot fix?

**Comment by intcreator** [comment]
I'm losing hope for a release within the next while, so I'm just setting up my dependencies on the master branch.  It works fine for me.

**Comment by devinivy** [comment]
@addyosmani Is there any chance this fix could also be ported into v0.6, particularly for apps stuck on Polymer v0.5?

**Comment by ssshake** [comment]
this just died over a year ago?
